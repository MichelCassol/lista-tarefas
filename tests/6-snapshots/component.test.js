const { renderUserCard, renderNotificationsList } = require('./component');

describe('Testes de Snapshot', () => {
  describe('renderUserCard', () => {
    test('deve renderizar corretamente com dados completos', () => {
      const user = {
        name: 'João Silva',
        email: 'joao@exemplo.com',
        avatar: '/images/avatar-1.jpg',
        role: 'Administrador',
        isActive: true
      };
      
      const output = renderUserCard(user);
      
      // Este snapshot será criado na primeira execução e comparado nas execuções seguintes
      expect(output).toMatchSnapshot();
    });
    
    test('deve renderizar com valores padrão quando alguns campos estão ausentes', () => {
      const user = {
        name: 'Maria Santos',
        email: 'maria@exemplo.com'
        // Sem avatar, role e isActive
      };
      
      const output = renderUserCard(user);
      expect(output).toMatchSnapshot();
    });
    
    test('deve renderizar usuário inativo', () => {
      const user = {
        name: 'Carlos Oliveira',
        email: 'carlos@exemplo.com',
        isActive: false
      };
      
      const output = renderUserCard(user);
      expect(output).toMatchSnapshot();
    });
    
    test('deve renderizar mensagem de erro para dados inválidos', () => {
      // Testes para entradas inválidas
      expect(renderUserCard(null)).toMatchSnapshot('usuário nulo');
      expect(renderUserCard(undefined)).toMatchSnapshot('usuário indefinido');
      expect(renderUserCard('string')).toMatchSnapshot('usuário como string');
    });
  });
  
  describe('renderNotificationsList', () => {
    test('deve renderizar uma lista de notificações', () => {
      const notifications = [
        { id: 1, message: 'Nova mensagem recebida', type: 'info', read: false },
        { id: 2, message: 'Pagamento confirmado', type: 'success', read: true },
        { id: 3, message: 'Atualização disponível', type: 'warning', read: false }
      ];
      
      const output = renderNotificationsList(notifications);
      expect(output).toMatchSnapshot();
    });
    
    test('deve renderizar mensagem quando não há notificações', () => {
      const output = renderNotificationsList([]);
      expect(output).toMatchSnapshot();
    });
    
    test('deve renderizar mensagem de erro para dados inválidos', () => {
      expect(renderNotificationsList(null)).toMatchSnapshot('notificações nulas');
      expect(renderNotificationsList('string')).toMatchSnapshot('notificações como string');
    });
    
    // Exemplo de uso de toMatchInlineSnapshot - snapshots são armazenados no próprio código
    test('deve renderizar notificações com valores padrão', () => {
      const notifications = [
        { id: 1, message: 'Notificação sem tipo ou status' }
        // Sem type e read
      ];
      
      const output = renderNotificationsList(notifications);
      
      // Em vez de criar um arquivo separado, o snapshot será inserido aqui na primeira execução
      // e nas próximas execuções será comparado com este snapshot inline
      expect(output).toMatchInlineSnapshot(`
"
    <ul class="notifications-list">
      
      <li class="notification-item unread" data-id="1">
        <div class="notification-item__icon notification-item__icon--info">
          <svg><!-- Ícone de Info --></svg>
        </div>
        <div class="notification-item__content">
          <p>Notificação sem tipo ou status</p>
        </div>
        <div class="notification-item__actions">
          <button class="btn btn--small">Marcar como lida</button>
        </div>
      </li>
    
    </ul>
  "
`);
    });
  });
  
  // Demonstração de snapshot de objeto (não apenas HTML)
  describe('Snapshots para estruturas de dados', () => {
    test('deve corresponder ao snapshot do objeto de usuário transformado', () => {
      const user = {
        name: 'João Silva',
        email: 'joao@exemplo.com',
        role: 'Usuário'
      };
      
      // Transforma os dados para um formato específico
      const transformedUser = {
        displayName: user.name.toUpperCase(),
        contactInfo: {
          email: user.email,
          phoneNumber: 'N/A'
        },
        permissions: ['read', 'write'],
        metadata: {
          role: user.role,
          lastLogin: new Date('2023-01-01').toISOString()
        }
      };
      
      // Podemos fazer snapshot de objetos também, não apenas de strings HTML
      expect(transformedUser).toMatchSnapshot();
    });
  });
}); 