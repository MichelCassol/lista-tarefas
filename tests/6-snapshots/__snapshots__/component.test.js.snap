// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes de Snapshot Snapshots para estruturas de dados deve corresponder ao snapshot do objeto de usuário transformado 1`] = `
{
  "contactInfo": {
    "email": "joao@exemplo.com",
    "phoneNumber": "N/A",
  },
  "displayName": "JOÃO SILVA",
  "metadata": {
    "lastLogin": "2023-01-01T00:00:00.000Z",
    "role": "Usuário",
  },
  "permissions": [
    "read",
    "write",
  ],
}
`;

exports[`Testes de Snapshot renderNotificationsList deve renderizar mensagem de erro para dados inválidos: notificações como string 1`] = `"<div class="error">Formato de notificações inválido</div>"`;

exports[`Testes de Snapshot renderNotificationsList deve renderizar mensagem de erro para dados inválidos: notificações nulas 1`] = `"<div class="error">Formato de notificações inválido</div>"`;

exports[`Testes de Snapshot renderNotificationsList deve renderizar mensagem quando não há notificações 1`] = `"<div class="empty-list">Não há notificações</div>"`;

exports[`Testes de Snapshot renderNotificationsList deve renderizar uma lista de notificações 1`] = `
"
    <ul class="notifications-list">
      
      <li class="notification-item unread" data-id="1">
        <div class="notification-item__icon notification-item__icon--info">
          <svg><!-- Ícone de Info --></svg>
        </div>
        <div class="notification-item__content">
          <p>Nova mensagem recebida</p>
        </div>
        <div class="notification-item__actions">
          <button class="btn btn--small">Marcar como lida</button>
        </div>
      </li>
    
      <li class="notification-item read" data-id="2">
        <div class="notification-item__icon notification-item__icon--success">
          <svg><!-- Ícone de Sucesso --></svg>
        </div>
        <div class="notification-item__content">
          <p>Pagamento confirmado</p>
        </div>
        <div class="notification-item__actions">
          <button class="btn btn--small">Marcar como não lida</button>
        </div>
      </li>
    
      <li class="notification-item unread" data-id="3">
        <div class="notification-item__icon notification-item__icon--warning">
          <svg><!-- Ícone de Aviso --></svg>
        </div>
        <div class="notification-item__content">
          <p>Atualização disponível</p>
        </div>
        <div class="notification-item__actions">
          <button class="btn btn--small">Marcar como lida</button>
        </div>
      </li>
    
    </ul>
  "
`;

exports[`Testes de Snapshot renderUserCard deve renderizar com valores padrão quando alguns campos estão ausentes 1`] = `
"
    <div class="user-card active">
      <div class="user-card__avatar">
        <img src="/images/default-avatar.png" alt="Maria Santos">
      </div>
      <div class="user-card__info">
        <h3 class="user-card__name">Maria Santos</h3>
        <p class="user-card__email">maria@exemplo.com</p>
        <span class="user-card__role">Usuário</span>
      </div>
      <div class="user-card__status">
        <span class="status-badge status-badge--active">Ativo</span>
      </div>
    </div>
  "
`;

exports[`Testes de Snapshot renderUserCard deve renderizar corretamente com dados completos 1`] = `
"
    <div class="user-card active">
      <div class="user-card__avatar">
        <img src="/images/avatar-1.jpg" alt="João Silva">
      </div>
      <div class="user-card__info">
        <h3 class="user-card__name">João Silva</h3>
        <p class="user-card__email">joao@exemplo.com</p>
        <span class="user-card__role">Administrador</span>
      </div>
      <div class="user-card__status">
        <span class="status-badge status-badge--active">Ativo</span>
      </div>
    </div>
  "
`;

exports[`Testes de Snapshot renderUserCard deve renderizar mensagem de erro para dados inválidos: usuário como string 1`] = `"<div class="error">Dados de usuário inválidos</div>"`;

exports[`Testes de Snapshot renderUserCard deve renderizar mensagem de erro para dados inválidos: usuário indefinido 1`] = `"<div class="error">Dados de usuário inválidos</div>"`;

exports[`Testes de Snapshot renderUserCard deve renderizar mensagem de erro para dados inválidos: usuário nulo 1`] = `"<div class="error">Dados de usuário inválidos</div>"`;

exports[`Testes de Snapshot renderUserCard deve renderizar usuário inativo 1`] = `
"
    <div class="user-card inactive">
      <div class="user-card__avatar">
        <img src="/images/default-avatar.png" alt="Carlos Oliveira">
      </div>
      <div class="user-card__info">
        <h3 class="user-card__name">Carlos Oliveira</h3>
        <p class="user-card__email">carlos@exemplo.com</p>
        <span class="user-card__role">Usuário</span>
      </div>
      <div class="user-card__status">
        <span class="status-badge status-badge--inactive">Inativo</span>
      </div>
    </div>
  "
`;
